### PR TITLE
gh-109287: Desugar inst(X) to op(X); macro(X) = X

### PR DESCRIPTION
**Pull request by gvanrossum**
I'm happy with this. The code generator needs to be cleaned up a bit more (e.g. the stats about instrs read/written are weird now).

* Issue: gh-109287

### CODE REVIEW

**Comment by gvanrossum** [comment]
Lots of tests failing or skipped, apparently due to test_threading being INTERRUPTED. Locally reproduced on macOS after rebase. Not sure yet where to look -- it seems to happen in test_interrupt_main_mainthread. I'm guessing I lost a CHECK_EVAL_BREAKER() call somewhere.

**Comment by gvanrossum** [comment]
Remaining test failures are caused by:

- missing support for super-instructions (test_capi)
- totally different generated output (test_generated_cases)

**Comment by gvanrossum** [comment]
I think super-instructions now work again (the old special-casing was salvageable). I have a few other things:

- [ ] Generate instructions in the same order as before.
- [ ] With `-Xuops` I get crashes in `importlib/_bootstrap.py`.
- [ ] Make `test_generated_cases` work.
- [ ] Clean up the code generator.

**Comment by gvanrossum** [comment]
I'd add @markshannon but he said he's on vacation until end of next week.

**Comment by gvanrossum** [comment]
CC @Fidget-Spinner (because I had to change the abstract interpretation code generator a bit).

File: Tools/cases_generator/analysis.py [hunk 1]
```diff
@@ -97,10 +97,7 @@ def parse(self) -> None:
         n_instrs = 0
         n_ops = 0
         for instr in self.instrs.values():
-            if instr.kind == "op":
-                n_ops += 1
-            else:
-                n_instrs += 1
+            n_ops += 1
```
Comment (iritkatriel) [inline Tools/cases_generator/analysis.py]: `n_ops = len(self.instrs)`

File: Tools/cases_generator/generate_cases.py [hunk 2]
```diff
@@ -255,15 +252,13 @@ def assign_opcode_ids(self) -> None:
...
         for name, family in self.families.items():
             specialized_ops.update(family.members)
 
-        for instr in itertools.chain(
-            [instr for instr in self.instrs.values() if instr.kind != "op"],
-            self.macro_instrs.values(),
-        ):
+        for instr in self.macro_instrs.values():
             assert isinstance(instr, (Instruction, MacroInstruction, PseudoInstruction))
+            assert isinstance(instr, MacroInstruction)
```
Comment (iritkatriel) [inline Tools/cases_generator/generate_cases.py]: The previous assert should be removed then.

**Pull request merged**